e2e/app/solvernetrouting.go

```diff
@@ -0,0 +1,148 @@
+		if err != nil {
+			return errors.Wrap(err, "configure inbox", "chain", chain.Name)
+		}
+
+		err = configureOutbox(ctx, backend, txOpts, addrs.SolverNetOutbox, remoteChainIDs, remoteInboxConfigs)
+		if err != nil {
+			return errors.Wrap(err, "configure outbox", "chain", chain.Name)
+		}
+	}
+
+	return nil
+}
+
+// getRoutes returns the remote chain IDs, outboxes, and inbox configs for a given chain.
+func getRoutes(chain netconf.Chain, allChains []netconf.Chain, inbox common.Address, outbox common.Address) ([]uint64, []common.Address, []bindings.ISolverNetOutboxInboxConfig, error) {
+	var remoteChainIDs []uint64
+	var remoteOutboxes []common.Address
+	var remoteInboxConfigs []bindings.ISolverNetOutboxInboxConfig
+
+	for _, c := range allChains {
+		if c.ID != chain.ID {
+			var solverNetInbox common.Address
+			var solverNetOutbox common.Address
+			provider := solvernet.None
+			var err error
+
+			if !solvernet.IsDisabled(chain.ID) && !solvernet.IsDisabled(c.ID) {
+				solverNetInbox = inbox
+				solverNetOutbox = outbox
+				provider, err = solvernet.Provider(c.ID)
+				if err != nil {
+					return nil, nil, nil, errors.Wrap(err, "get provider", "chain", c.Name)
+				}
+			}
+
+			remoteChainIDs = append(remoteChainIDs, c.ID)
+			remoteOutboxes = append(remoteOutboxes, solverNetOutbox)
+			remoteInboxConfigs = append(remoteInboxConfigs, bindings.ISolverNetOutboxInboxConfig{
+				Inbox:    solverNetInbox,
+				Provider: provider,
+			})
+		}
+	}
+
+	return remoteChainIDs, remoteOutboxes, remoteInboxConfigs, nil
+}
+
+// isDeployed returns true if the SolverNet inbox and outbox are deployed on a given chain.
+func isDeployed(ctx context.Context, backend *ethbackend.Backend, inbox common.Address, outbox common.Address) (bool, error) {
+	isDeployed, err := contracts.IsDeployed(ctx, backend, inbox)
+	if !isDeployed || err != nil {
+		return false, errors.Wrap(err, "solvernetinbox not deployed", "chain", backend.Name())
+	}
+
+	isDeployed, err = contracts.IsDeployed(ctx, backend, outbox)
+	if !isDeployed || err != nil {
+		return false, errors.Wrap(err, "solvernetoutbox not deployed", "chain", backend.Name())
+	}
+
+	return true, nil
+}
+
+// configureInbox configures the routes on a SolverNetInbox contract for a given chain.
+func configureInbox(ctx context.Context, backend *ethbackend.Backend, txOpts *bind.TransactOpts, inbox common.Address, remoteChainIDs []uint64, remoteOutboxes []common.Address) error {
+	solverNetInbox, err := bindings.NewSolverNetInbox(inbox, backend)
+	if err != nil {
+		return errors.Wrap(err, "bind inbox", "chain", backend.Name())
+	}
+
+	tx, err := solverNetInbox.SetOutboxes(txOpts, remoteChainIDs, remoteOutboxes)
+	if err != nil {
+		return errors.Wrap(err, "set outboxes on inbox", "chain", backend.Name())
+	}
+
+	_, err = backend.WaitMined(ctx, tx)
+	if err != nil {
+		return errors.Wrap(err, "wait mined set outboxes on inbox", "chain", backend.Name())
+	}
```

Comment:
try to have shorter error messages. You'll note this is a pattern we follow elsewhere. We want messages to be precise enough where we can follow wrapped errors to their source, but not more verbose thent hat

```golang
tx, err := solverNetInbox.SetOutboxes(txOpts, remoteChainIDs, remoteOutboxes)
	if err != nil {
		return errors.Wrap(err, "set outboxes", "chain", backend.Name())
	}

	_, err = backend.WaitMined(ctx, tx)
	if err != nil {
		return errors.Wrap(err, "wait mined", "chain", backend.Name())
	}
```